packages/fxa-auth-server/lib/routes/auth-schemes/hawk-fxa-token.js

```diff
@@ -0,0 +1,145 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const AppError = require('../../error');
+const Boom = require('@hapi/boom');
+
+// The following regexes and Hawk header parsing are taken from the Hawk library.
+// See https://github.com/mozilla/hawk/blob/01f3d35479fe76654bb50f2886b37310555d088e/lib/utils.js#L126
+const authHeaderRegex = /^(\w+)(?:\s+(.*))?$/; // Header: scheme[ something]
+const attributeRegex = /^[ \w!#$%&'()*+,\-./:;<=>?@[\]^`{|}~]+$/; // !#$%&'()*+,-./:;<=>?@[]^_`{|}~ and space, a-z, A-Z, 0-9
+function parseAuthorizationHeader(header, keys) {
+  keys = keys || ['id', 'ts', 'nonce', 'hash', 'ext', 'mac', 'app', 'dlg'];
+
+  if (!header) {
+    throw Boom.unauthorized(null, 'Hawk');
+  }
+
+  if (header.length > 4096) {
+    throw Boom.badRequest('Header length too long');
+  }
+
+  const headerParts = header.match(authHeaderRegex);
+  if (!headerParts) {
+    throw Boom.badRequest('Invalid header syntax');
+  }
+
+  const scheme = headerParts[1];
+  if (scheme.toLowerCase() !== 'hawk') {
```

Comment:
This function wouldn't get called if the header started with `hawk` (lowercase h) due to the logic on line 97 (`auth.indexOf('Hawk') > -1`). So either the toLowerCase here is unnecessary, or line 97 is not correct?